src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/LifeCycle/LifeCycleEffect.shared.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Xamarin.CommunityToolkit.Helpers;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Effects
+{
+	/// <summary>
+	/// An effect to subscribe to the View's lifecycle events.
+	/// </summary>
+	public class LifecycleEffect : RoutingEffect
+	{
+		readonly WeakEventManager eventManager = new WeakEventManager();
+
+		/// <summary>
+		/// Event that is triggered when the <see cref="View" /> is loaded and is ready for use.
+		/// </summary>
+		public event EventHandler Loaded
+		{
+			add => eventManager.AddEventHandler(value);
+			remove => eventManager.RemoveEventHandler(value);
+		}
+
+		/// <summary>
+		/// Event that is triggered when the <see cref="View" /> is unloaded and isn't ready for use.
+		/// </summary>
+		public event EventHandler Unloaded
+		{
+			add => eventManager.AddEventHandler(value);
+			remove => eventManager.RemoveEventHandler(value);
+		}
+
+		/// <summary>
+		/// Constructor for the <see cref="LifecycleEffect" />
+		/// </summary>
+		public LifecycleEffect()
+			: base(EffectIds.LifeCycleEffect)
+		{
+#if __ANDROID__
+			if (System.DateTime.Now.Ticks < 0)
+				_ = new Xamarin.CommunityToolkit.Android.Effects.LifeCycleEffectRouter();
+#elif __IOS__
+			if (System.DateTime.Now.Ticks < 0)
+				_ = new Xamarin.CommunityToolkit.iOS.Effects.LifeCycleEffectRouter();
+#elif UWP
+			if (System.DateTime.Now.Ticks < 0)
+				_ = new Xamarin.CommunityToolkit.UWP.Effects.LifeCycleEffectRouter();
+#endif
```

Comment:
Just curious - how come we're initializing an object and not assigning it to anything? Is to prevent the linker from removing the platform-specific implementation?

Reply:
Now that I think about it, it's gotta be for the linker, because `System.DateTime.Now.Ticks < 0` will never be true, right? 😃

Reply:
Maybe we wrap this in a `#region` and add a comment about this being for the linker to prevent devs in the future from removing this block of code?

```csharp
#region Add work-around to prevent linker from removing the platform-specific implementation 

#if __ANDROID__
			if (System.DateTime.Now.Ticks < 0)
				_ = new Xamarin.CommunityToolkit.Android.Effects.LifeCycleEffectRouter();
#elif __IOS__
			if (System.DateTime.Now.Ticks < 0)
				_ = new Xamarin.CommunityToolkit.iOS.Effects.LifeCycleEffectRouter();
#elif UWP
			if (System.DateTime.Now.Ticks < 0)
				_ = new Xamarin.CommunityToolkit.UWP.Effects.LifeCycleEffectRouter();
#endif

#endregion

Reply:
@brminnick  I remember I have asked the same question in the past and @AndreiMisiukevich answered me that is to trick the linker. to preserver, indeed like you thought :).

Reply:
Yes @brminnick this is a trick to do a static reference to make linker don't remove the native implementation, and it will not remove just if someone uses the effect. And the compiler will do optimization on this part because this will never be called. I learned it from [here](https://github.com/xamarin/Essentials/pull/1209#discussion_r405051562).

@brminnick about the regions, do you agree to do it in another PR? That way we can do it for all effects

Reply:
> do you agree to do it in another PR? That way we can do it for all effects 

No problem!

I'll open an Issue now and assign it to myself so that I don't forget 👍

Reply:
https://github.com/xamarin/XamarinCommunityToolkit/issues/910 ✅